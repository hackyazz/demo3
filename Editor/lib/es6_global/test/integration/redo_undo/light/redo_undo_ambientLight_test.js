

import * as Curry from "../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as RedoUndoTool$WonderEditor from "../tool/RedoUndoTool.js";
import * as PickColorTool$WonderEditor from "../../../tool/PickColorTool.js";
import * as ControllerTool$WonderEditor from "../../../unit/composable_component/controller/tool/ControllerTool.js";
import * as BuildCanvasTool$WonderEditor from "../../../tool/BuildCanvasTool.js";
import * as EventListenerTool$WonderEditor from "../../../unit/tool/EventListenerTool.js";
import * as MainEditorSceneTool$WonderEditor from "../../../tool/MainEditorSceneTool.js";
import * as BuildComponentForCurryTool$WonderEditor from "../../../tool/BuildComponentForCurryTool.js";

describe("redo_undo: ambientLight", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        var _changeColorAndPushUndoStack = function (value) {
          var sourceColor = ControllerTool$WonderEditor.getColor(/* () */0);
          ControllerTool$WonderEditor.changeColor(value);
          return ControllerTool$WonderEditor.closeColorPicker(sourceColor, undefined, undefined, /* () */0);
        };
        var _simulateTwiceChangeAmbientLight = function () {
          BuildCanvasTool$WonderEditor.buildFakeCanvas(sandbox);
          var color1 = PickColorTool$WonderEditor.buildColor1(/* () */0);
          var color2 = PickColorTool$WonderEditor.buildColor2(/* () */0);
          _changeColorAndPushUndoStack(color1);
          return _changeColorAndPushUndoStack(color2);
        };
        var _beforeEach = function () {
          return MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, (function () {
                        return /* () */0;
                      }));
        };
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
                return Curry._1(EventListenerTool$WonderEditor.stubGetElementByIdReturnFakeDom, Curry._1(EventListenerTool$WonderEditor.buildFakeDom, /* () */0));
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        return RedoUndoTool$WonderEditor.testRedoUndoTwoStep(sandbox, "prepare first step: set currentSceneTreeNode", /* tuple */[
                    _simulateTwiceChangeAmbientLight,
                    _beforeEach,
                    (function () {
                        return /* () */0;
                      })
                  ], BuildComponentForCurryTool$WonderEditor.buildHeader);
      }));

export {
  
}
/*  Not a pure module */
