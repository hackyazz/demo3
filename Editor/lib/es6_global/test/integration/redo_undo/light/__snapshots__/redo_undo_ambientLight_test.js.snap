// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: ambientLight prepare first step: set currentSceneTreeNode test redo operate test redo one step undo step which from second to zero, redo step which from zero to first 1`] = `
<article
  className="wonder-header-component"
>
  <div
    className="header-nav"
  >
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          File
        </span>
      </div>
    </div>
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          Edit
        </span>
      </div>
    </div>
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          Publish
        </span>
      </div>
    </div>
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          Help
        </span>
      </div>
    </div>
  </div>
</article>
`;

exports[`redo_undo: ambientLight prepare first step: set currentSceneTreeNode test redo operate test redo two step undo step which from second to zero,redo step which from zero to second 1`] = `
<article
  className="wonder-header-component"
>
  <div
    className="header-nav"
  >
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          File
        </span>
      </div>
    </div>
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          Edit
        </span>
      </div>
    </div>
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          Publish
        </span>
      </div>
    </div>
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          Help
        </span>
      </div>
    </div>
  </div>
</article>
`;

exports[`redo_undo: ambientLight prepare first step: set currentSceneTreeNode test undo operate test not undo 1`] = `
<article
  className="wonder-header-component"
>
  <div
    className="header-nav"
  >
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          File
        </span>
      </div>
    </div>
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          Edit
        </span>
      </div>
    </div>
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          Publish
        </span>
      </div>
    </div>
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          Help
        </span>
      </div>
    </div>
  </div>
</article>
`;

exports[`redo_undo: ambientLight prepare first step: set currentSceneTreeNode test undo operate test undo one step step which from second to first 1`] = `
<article
  className="wonder-header-component"
>
  <div
    className="header-nav"
  >
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          File
        </span>
      </div>
    </div>
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          Edit
        </span>
      </div>
    </div>
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          Publish
        </span>
      </div>
    </div>
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          Help
        </span>
      </div>
    </div>
  </div>
</article>
`;

exports[`redo_undo: ambientLight prepare first step: set currentSceneTreeNode test undo operate test undo two step step which from second to zero 1`] = `
<article
  className="wonder-header-component"
>
  <div
    className="header-nav"
  >
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          File
        </span>
      </div>
    </div>
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          Edit
        </span>
      </div>
    </div>
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          Publish
        </span>
      </div>
    </div>
    <div
      className="header-item"
    >
      <div
        className="component-item"
      >
        <span
          className="item-title"
          onClick={[Function]}
          onMouseOver={[Function]}
        >
          Help
        </span>
      </div>
    </div>
  </div>
</article>
`;
